packages/@aws-cdk/aws-ec2/lib/instance-types.ts

```diff
@@ -332,6 +332,21 @@ export enum InstanceClass {
    */
   D2 = 'd2',
 
+  /**
+   * Storage-optimized instances, 3rd generation
+   */
+  STORAGE3 = 'd3',
+
+  /**
+   * Storage-optimized instances, 3rd generation
+   */
+  D3 = 'd3',
+
+  /**
+  * Storage-optimized instances, 3rd generation
+  */
+  D3EN = 'd3en',
```

Comment:
Could you please add a symbolic name for this enum, too? Thanks.

Reply:
Can you suggest one?

Reply:
The difference between the two instance types is described on the [product page](https://aws.amazon.com/de/ec2/instance-types/d3/): 
> D3en instances offer a 3.5x higher storage-to-vCPU ratio compared to D2 instances.

Maybe we could use `STORAGE3_ENHANCED_STORAGE` or just `STORAGE3_ENHANCED`?

Or does someone else have a suggestion?

Reply:
@njlynch wdyt?

Reply:
I'm a bit torn on having the longer/symbolic names, given that we effectively just make up the names for the suffixes, and there's no consistency or guidelines. That being said, we've started down this road, so we might as well continue for the time being.

It's hard to pin down what `en` stands for from the blog post, but I _think_ it stands for "elastic networking".

> The D3en instances have a high ratio of storage to vCPU, and are optimized for high throughput and high sequential I/O to very large data sets, with a cost-per-TB that is 80% lower than on D2 instances.
> AMIs – You must use AMIs that include the Elastic Network Adapter (ENA) and NVMe drivers.

That's my guess from the above, at least. That would imply `STORAGE3_ELASTIC_NETWORK` would be the name.

Reply:
Ah, this seems to be the official documentation on the suffixes: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#instance-hardware-specs

So:
* `e – Extra storage or memory`
* `n – Network optimization`

`STORAGE3_ENHANCED_NETWORK` might be a decent pick. Or `STORAGE3_EXTRA_STORAGE_NETWORK`.

Reply:
Personally I never use symbolic names :D. Thanks for the input @njlynch. PR has been updated.